File: ir-core/src/main/kotlin/com/voxeldev/mcodegen/dsl/ir/IrLocation.kt
package com.voxeldev.mcodegen.dsl.ir

import kotlinx.serialization.Serializable

/**
 * Represents a location in the source code in the IR (Intermediate Representation) system.
 * This class contains information about where an element is defined in the source code,
 * including file path, line number, and column number.
 */
@Serializable
data class IrLocation(
    val filePath: String,
    val lineNumber: Int,
    val columnNumber: Int
)

File: swift-companion/Sources/IrCore/IrLocation.swift
//
//  IrLocation.swift
//  swift-companion
//
//  Created by nVoxel on 17.06.2025.
//

import Foundation

public struct IrLocation: Codable {
    public let type: String?
    public let filePath: String
    public let lineNumber: Int
    public let columnNumber: Int

    public init(filePath: String, lineNumber: Int, columnNumber: Int) {
        self.type = "com.voxeldev.mcodegen.dsl.ir.IrLocation"
        self.filePath = filePath
        self.lineNumber = lineNumber
        self.columnNumber = columnNumber
    }
}
